token/request.go

```diff
@@ -873,6 +1003,14 @@ func (t *Request) countOutputs() (int, error) {
 }
 
 func (t *Request) parseInputIDs(inputs []*token.ID) ([]*token.ID, token.Quantity, string, error) {
+	if t.TokenService == nil || t.TokenService.Vault() == nil || t.TokenService.Vault().NewQueryEngine() == nil {
```

Comment:
shall avoid this?

Reply:
I agree - I checked the code. This should never happens.